pywbem_mock/_instancewriteprovider.py

```diff
@@ -270,367 +209,280 @@ def __init__(self, cimrepository=None):
+                creation_class, new_instance, namespace=namespace, strict=True)
+        except ValueError as exc:
+            raise CIMError(CIM_ERR_INVALID_PARAMETER, str(exc))
 
-        self.validate_namespace(namespace)
+        # Get the instance store of the CIM repository. Since the existence of
+        # the namespace has already been verified, this will always succeed.
         instance_store = self.cimrepository.get_instance_store(namespace)
 
-        # Requires corresponding class to build path to be returned
-        target_class = self.get_class(namespace,
-                                      new_instance.classname,
-                                      local_only=False,
-                                      include_qualifiers=True,
-                                      include_classorigin=True)
-
-        # Test all key properties in instance. The CIM repository
-        # cannot add values for key properties and does
-        # not allow creating key properties from class defaults.
-        key_props = [p.name for p in six.itervalues(target_class.properties)
-                     if 'key' in p.qualifiers]
-        for pn in key_props:
-            if pn not in new_instance:
-                raise CIMError(
-                    CIM_ERR_INVALID_PARAMETER,
-                    _format("Key property {0!A} not in NewInstance ", pn))
-
-        # Exception if property in instance but not class or types do not
-        # match
-        validate_inst_props(namespace, target_class, new_instance)
-
-        # Build instance path from values in the instance.
-        new_instance.path = CIMInstanceName.from_instance(
-            target_class,
-            new_instance,
-            namespace=namespace)
-
-        # Store the new instance in the  CIM repository if it is not
-        # already in the repository
+        # Store the new instance in the CIM repository, verifying that it does
+        # not exist yet.
         try:
             instance_store.create(new_instance.path, new_instance)
         except ValueError:
             raise CIMError(
                 CIM_ERR_ALREADY_EXISTS,
-                _format("NewInstance {0!A} already exists in namespace "
+                _format("New instance {0!A} already exists in namespace "
                         "{1!A}.", new_instance.path, namespace))
 
-        # Create instance returns model path
+        # CreateInstance returns the instance path of the new instance
         return new_instance.path
 
     ####################################################################
     #
-    #   ModifyInstance server request
+    #   ModifyInstance mock WBEM server responder
     #
     ####################################################################
 
-    def ModifyInstance(self, ModifiedInstance,
-                       IncludeQualifiers=None, PropertyList=None):
+    def ModifyInstance(self, modified_instance, IncludeQualifiers=None):
```

Comment:
Doesn't removing the property list make us non-compliant.  I agree that it was not worth much but I actually used it as an alternative to removing properties when, for example, I wanted to just change the value of a single property.  (get from server, modify property value, set PropertyList) so you don't have to worry about key properites since you just send the whole thing back with a propertylist.

Reply:
Ahha.  I did not look yet but does this mean that you cleaned them in the dispatcher so the property list no longer needed.